app/main.py

```diff
@@ -1,8 +1,21 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_of_people = []
+    for person in people:
+        list_of_people.append(Person(person["name"], person["age"]))
+
+    for i, person in enumerate(people):
+        if "wife" in person and person["wife"] is not None:
+            list_of_people[i].wife = Person.people[person["wife"]]
```

Comment:
You could use the `get` method and walrus operator to reduce the amount of checks and store the retrieved value:
```suggestion
        if wife_name := person.get("wife"):
            list_of_people[i].wife = Person.people[wife_name]
```
Apply this change to two conditions you have.